### DISCUSSION TITLE
Realtime support: websockets/turbo streams/anycable

### DISCUSSION THREAD

**Discussion by synth** [yatish27/shore#14] [Ideas]
Hey Yatish! 

Saw your talk at the SF Ruby Meetup :) The stack in this base app is awesome and something I've been scoping to find for a while. 

I'm keen to use it for an app I'm building which will be heavily realtime.  Have you thought about adding Turbo Streams/AnyCable to this base app? I did a POC for my app last year and ran into a few issues struggling with TurboStreams and AnyCable. I quickly ran into issues with stock TurboStreams and then integrated AnyCable and struggled to get aspects like authenticated websockets to work correctly. I know AnyCable has had a lot of improvements since, and I'm getting back to the app I want to build starting fresh, and considering Shore for the greenfield setup. 

If you're interested, I'm down to explore this by building on top of this base app and can contribute back to this repo. LMKWYT.

**Comment by yatish27** [comment]
TurboStreams/AnyCable/ActionCable is not used in most apps. It would add additional bloat on the base app.
But I do like the idea of having base apps based on use cases. We can add this and expose it as a different branch.

Feel free to create a PR and we can collaborate

**Comment by synth** [comment]
Awesome sounds good! Yea, I've started down a path on a particular use case and made some decisions like using Devise for auth (which i'm not thrilled about, but seems to have the broadest usage and documentation).

**Comment by yatish27** [comment]
> which i'm not thrilled about

What is the issue?

**Comment by synth** [comment]
Oh, it's just that I've never been a fan of the heavy handed approach Devise took for auth. But it's fine :) What's your preferred auth?

**Reply by yatish27** [reply]
Devise is tried and tested, so I use it for most of my projects
